### PR TITLE
ES Translations, localized hard-coded strings and other improvements.

### PR DESCRIPTION
**Pull request by Deses**
I wanted to translate RDM to Spanish, and then I found out that there were many hard-coded string for the pop-ups and the stats page. Also, the date picker plugin wasn't using localizing so I added it.

## Description
--- Translations:
* Extracted all the hard-coded string from index.js to the lang files and replaced them with the respective i18n function.
* Updates EN and DE files with the new lines.
* New ES file with Spanish strings.
* Changed weathertypes,json into a new format that allows for localization.
* Added missing localizations to the stats page and made Flatpickr use the locale
--- Improvements:
* Forced initMap() to wait for all getJson() functions to complete before proceeding.
* Made popups with a minimal width to prevent elements overlapping.
* Changed popup logic to not process popups that are hidden. When shown again, they update correctly.
*

### Why is this change required?
I wanted RDM in Spanish. :P

### What problem does it solve?
RDM did not have a Spanish translation and many strings were hard-coded.

### If it fixes an open issue, please link to the issue here.
No issue.

### Please describe in detail how you tested your changes.
Since I'm using docker, to test the changes I made to the .swift files, I cloned my fork, built it locally and uploaded a package to ghcr.io and then pulled it into my docker containers.

```
rdm:
    image: ghcr.io/deses/realdevicemap:translations
    ....
```

To quickly test the changes made in the js and lang files, I bound the webroot folder to a folder accesible to me and placed the files there. `- ./rdm:/app/resources/webroot/`

### Include details of your testing environment, and the tests you ran to
All ran in a Linux Mint 20.3 VM and then I just manually used RDM to see the changes.

### See how your change affects other areas of the code, etc.
It shouldn't affect anything!

## Screenshots (if appropriate):
image
image
image
image
image
image
image

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Deses** [comment]
Since I made so many commits, I recommend you "squash and merge pr" rather than just merge it as to not pollute too much git's history. 😅

File: resources/webroot/index.js.mustache [hunk 1]
```diff
@@ -2554,12 +2566,23 @@ function getPokestopPopupContent (pokestop) {
...
+      let choppedName = new Array();
+
+      while ((m = regex.exec(pokestop.name)) !== null) {
+        choppedName.push(m[0]);
+      }
+      pokestopName = choppedName.join('<br>')
+    }
   }
 
   content += '<span class="text-nowrap"><h6><b>{{#perm_admin}}<a href="#" onclick="editPokestopName(\''+pokestop.id+'\',\''+pokestop.name+'\')">{{/perm_admin}}' +
```
Comment (Fabio1988) [inline resources/webroot/index.js.mustache:2588]: I think we can keep the old way for pokestop/gym name and remove `text-nowrap` for name. maybe use 2 cols for team icon and name :)

**Review by Fabio1988 — CHANGES_REQUESTED** [review]

File: resources/webroot/static/data/de.json [hunk 2]
```diff
@@ -12,15 +12,100 @@
...
+        "map_wind_level": "Wind Level",
+        "map_snow_level": "Schnee Level",
+        "map_fog_level": "Nebel Level",
+        "map_special_effeccts_level": "Spezialeffekte Level",
+        "map_severity": "Schweregrad",
+        "map_weather_warning": "Wetterwarnung",
+        "map_device_instance": "Instanz",
+        "map_device_last_seen": "Zuletzt gesehen",
+        "map_device_status": "Status",
+        "map_spawnpoint": "Spawnpunkte",
```
Comment (Fabio1988) [inline resources/webroot/static/data/de.json:99]: Spawnpunkt

Comment (Fabio1988) [inline resources/webroot/static/data/de.json:99]: Spawnpunkt

File: resources/webroot/static/data/de.json [hunk 3]
```diff
@@ -12,15 +12,100 @@
...
+        "map_despawn_timer": "Spawn-Zeiten",
```
Comment (Fabio1988) [inline resources/webroot/static/data/de.json:100]: `"map_despawn_timer": "Despawn-Zeit"`

Comment (Fabio1988) [inline resources/webroot/static/data/de.json:100]: `"map_despawn_timer": "Despawn-Zeit"`

File: resources/webroot/index.js.mustache [hunk 4]
```diff
@@ -58,6 +58,14 @@ let openedSubmissionTypeCell
 let openedWeather
 let openedDevice
 
+let popupPokemon
```
Comment (Fabio1988) [inline resources/webroot/index.js.mustache:61]: I think that is causing the direct link problems?! :+1:

**Review by Fabio1988 — APPROVED** [review]

**Review by Fabio1988 — APPROVED** [review]

**Pull request merged**